syndeo-template/src/main/java/com/google/cloud/syndeo/SyndeoTemplate.java

```diff
@@ -189,11 +193,55 @@ public static ConfiguredSchemaTransform buildFromJsonConfig(JsonNode transformCo
                                         ? fieldNode.asText()
                                         : fieldNode.isArray()
                                             ? new ObjectMapper().convertValue(fieldNode, List.class)
-                                            : new ObjectMapper().convertValue(fieldNode, Map.class))
+                                            : convertJsonNodeToRow(fieldNode))
             .collect(Collectors.toList());
-    return new ProviderUtil.TransformSpec(
+    // return new ProviderUtil.TransformSpec(
+    //         transformConfig.get("urn").asText(), configurationParameters)
+    //     .toProto();
+    ConfiguredSchemaTransform t = new ProviderUtil.TransformSpec(
             transformConfig.get("urn").asText(), configurationParameters)
         .toProto();
+
+    LOG.warn("configured schema transfrm " + t.getConfigurationValues().toString());
+    // LOG.warn("fiel " + t.getConfigurationValues().)
+    return t;
+  }
+
+  private static Row convertJsonNodeToRow(JsonNode node) {
```

Comment:
here you can probably get the `Field` object, and get the schema from it via `Field.getType().getRowSchema()` - that way you don't have to infer the field automatically.